cypress/support/step_definitions/timetable.js

```diff
@@ -1,10 +1,57 @@
-import { And, Then } from 'cypress-cucumber-preprocessor/steps';
+import { And, Then, When } from 'cypress-cucumber-preprocessor/steps';
+import { activity, appointment } from '../../mockApis/data';
+
+import { format, addDays } from 'date-fns';
 
 Then('I am displayed todays timetable', () => {
   cy.get('.todays-events > .govuk-body').contains(/.+/);
   cy.log('Timetable data has been passed through');
 });
 
-And('click the {string}', timetableLink => {
-  cy.get('a').contains(timetableLink).first().click();
+const daysFromNow = n => format(addDays(new Date(), n), 'yyyy-MM-dd');
+
+const tableToObject = table => {
+  const [keys, ...rows] = table.rawTable;
+  return rows.map(values =>
+    keys.reduce((o, k, i) => {
+      o[k] = values[i];
+      return o;
+    }, {}),
+  );
+};
+
+When('I have the following up coming events', args => {
+  const rows = tableToObject(args);
+
+  const events = rows.map(row => {
+    const date = row.when === 'today' ? daysFromNow(0) : daysFromNow(1);
+    return row.type === 'activity'
+      ? activity(date, row.start, row.end, row.name, row.location)
+      : appointment(date, row.start, row.end, row.name, row.location);
+  });
+
+  cy.task('stubEvents', events);
```

Comment:
Reads from the table of upcoming events and converts into the prison api response format.
Then tells wiremock to return these when the application asks for them.